general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -10,27 +10,94 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+# mypy: disable-error-code="empty-body, operator"
+
 """Client for making requests to Superstaq's API."""
 
 from __future__ import annotations
 
+import copy
+import enum
 import json
 import os
 import pathlib
 import sys
 import textwrap
 import time
 import urllib
+import uuid
 import warnings
 from collections.abc import Callable, Mapping, Sequence
-from typing import Any, ClassVar, TypeVar
+from typing import Any, ClassVar, Literal, TypeVar
 
 import requests
 
 import general_superstaq as gss
+from general_superstaq import _models
 
 TQuboKey = TypeVar("TQuboKey")
 
+RECOGNISED_CIRCUIT_TYPES = Literal[_models.CircuitType.CIRQ, _models.CircuitType.QISKIT]
+"""The circuit types that are currently implemented within the SuperstaqClient."""
+
+
+class _API_Version(str, enum.Enum):
+    V0_2_0 = "v0.2.0"
+    V0_3_0 = "v0.3.0"
+
+
+class _versioned_method:
```

Comment:
Camel space your class names.

Reply:
@vietphamngoc don't forget to camel space this (and `_API_Version` above)